src/main/kotlin/org/opensearch/replication/action/changes/TransportGetChangesAction.kt

```diff
@@ -173,7 +173,7 @@ class TransportGetChangesAction @Inject constructor(threadPool: ThreadPool, clus
     override fun shards(state: ClusterState, request: InternalRequest): ShardsIterator {
         val shardIt = state.routingTable().shardRoutingTable(request.request().shardId)
         // Random active shards
-        return if (ValidationUtil.isRemoteStoreEnabledCluster(clusterService)) shardIt.primaryShardIt()
+        return if (ValidationUtil.isRemoteStoreEnabledCluster(clusterService) || ValidationUtil.isRemoteMigrating(clusterService)) shardIt.primaryShardIt()
```

Comment:
I think we can combine the 2 methods into `isRemoteEnabledOrMigrating(clusterService)`

Reply:
I intentionally left them as a separate methods considering future changes which might require them individually.  Let me know if we don't need it anymore, i can go ahead and combine them. cc: @gbbafna

Reply:
Let's unify those for now. `isRemoteStoreEnabledCluster` can still be a separate method in `ValidationsUtil`.

Reply:
Sure